### PR TITLE
first draft of java11 compat support

### PR DESCRIPTION
**Pull request by lhein**
Signed-off-by: Lars Heinemann <lhein.smx@gmail.com>

# Pull Request Checklist

After this checklist is all checked or PR provides explanations for possible pass-through, please put the label "Ready for review" on the PR.

## General

- [ ] Did you use the Jira Issue number in the commit comments?
- [ ] Did you set meaningful commit comments on each commit?
- [ ] Did you sign off all commits for this PR? (git commit -s -m "jira issue number - your commit comment")

## Functional

- [ ] Did the CI job report a successful build?
- [ ] Is the non-happy path working, too?

## Maintainability

- [ ] Are all Sonar reported issues fixed or can they be ignored?
- [ ] Is there no duplicated code?
- [ ] Are method-/class-/variable-names meaningful?

## Tests

- [ ] Are there unit-tests?
- [ ] Are there integration tests (or at least a jira to tackle these)?
- [ ] Do we need a new UI test?

## Legal

- [ ] Have you used the correct file header copyright comment?
- [ ] Have you used the correct year in the headers of new files?

### CODE REVIEW

**Comment by apupier** [comment]
@jeffmaury thanks for the help. Can you signed-off your commit please?

**Comment by jeffmaury** [comment]
Did not find out how to do from the Github web UI

**Comment by apupier** [comment]
> Did not find out how to do from the Github web UI

I don't think that you can from web UI.
You need to modify the commit message to include the signed-off-by and then to force push the branch

File: core/plugins/org.fusesource.ide.camel.model.service.core/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -29,7 +29,8 @@ Require-Bundle: org.eclipse.core.resources;bundle-version="3.10.0",
  com.fasterxml.jackson.core.jackson-databind;bundle-version="2.5.0",
  org.fusesource.ide.foundation.core,
  org.fusesource.ide.foundation.ui,
- org.eclipse.m2e.core.ui;bundle-version="1.6.2"
+ org.eclipse.m2e.core.ui;bundle-version="1.6.2",
+ org.jboss.tools.common.jaxb;resolution:=optional
```
Comment (apupier) [inline core/plugins/org.fusesource.ide.camel.model.service.core/META-INF/MANIFEST.MF]: using resolution optional means that with Java 11, if org.jboss.tools.common.jaxb bundle is not available it won't work.
When inside Jboss Tools, the plugin is part of it i think. I don't think that it is the case for vanilla Fuse Tooling. I think we need to add the dependency to this plugin in one of our Camel feature explicitely.

Comment (jeffmaury) [reply core/plugins/org.fusesource.ide.camel.model.service.core/META-INF/MANIFEST.MF]: Remove the optional one

Comment (apupier) [reply core/plugins/org.fusesource.ide.camel.model.service.core/META-INF/MANIFEST.MF]: sure that optional can be removed? i think that we will have 2 jaxb implementation when on Java 8. This is something that is not working as far as I know.

**Comment by lhein** [comment]
Could be that we need to import bundle javax.xml.bind @apupier ?

**Comment by lhein** [comment]
can you test again, @apupier ? My tests did not show an error.

**Comment by apupier** [comment]
> can you test again, @apupier ? My tests did not show an error.

still same error.

more precise steps:
- create a Fuse integration project with a ll defaults
- Drag'n'drop the "Data Transformation"
- use xml as source and java as target
- choose xml isntance document, and configureation/settings.xml file
- choose java.awt.desktop.AboutEvent class
- click Finish
--> the wizard stays opened and there is the previously provided stacktrace

**Comment by lhein** [comment]
I tested it with 2 xml documents and it had no errors...weird.

**Comment by lhein** [comment]
Tested with XML 2 Java transformation. Source settings.xml and target java.awt.Desktop (I was unable to find the class you mentioned)
 
Worked fine here without errors.

Bildschirmfoto von 2019-03-11 14-24-58

**Comment by tsedmik** [comment]
@djelinek could you please look at this PR. Thank you in advance.

**Comment by lhein** [comment]
Its broken for me too. I had still Java 8 selected in my launch config.

**Comment by apupier** [comment]
a draft to get pass the reported error:
https://github.com/apupier/jbosstools-fuse/pull/new/apupier-java11_wip
https://github.com/apupier/jbosstools-base/pull/new/java11

**Comment by lhein** [comment]
We should coordinate with @jeffmaury asap to get the changes into the jbosstools plugin and to respin the target platform build.

**Comment by apupier** [comment]
> We should coordinate with @jeffmaury asap to get the changes into the jbosstools plugin and to respin the target platform build.

I think there is no need to rebuild the Target Platform. It is not part of locus but it is part of Jboss Tools base.
Currently creating a dedicated plugin in our codebase to be independent of Jboss Tools availability

**Comment by jeffmaury** [comment]
Does it means there is no need for a JBoss Tools base change ? Created the PR but can delete it

**Comment by apupier** [comment]
> Does it means there is no need for a JBoss Tools base change ? Created the PR but can delete it

i think it is best to have it in Jboss Tools as it seems to be something really generic but as I didn't have news on mattermost and it is highest priority I was trying to maximize chance to have one way to have it working

**Comment by apupier** [comment]
The mentioned PR in JBoss Tools directly: https://github.com/jbosstools/jbosstools-base/pull/639

**Comment by apupier** [comment]
I created another branch https://github.com/apupier/jbosstools-fuse/pull/new/FUSETOOLS-2795-Java11Compatibility2 with last modifications required as soon as  build corresponding to the merge of jbosstools/jbosstools-base#639 will be available

**Pull request closed without merging**